src/v/storage/tests/CMakeLists.txt

```diff
@@ -50,14 +50,23 @@ rp_test(
   ARGS "-- -c 1"
 )
 
+rp_test(
+  UNIT_TEST
+  BINARY_NAME storage_log_index
```

Comment:
Nit: I'm trying to come up with a naming scheme such as `<component>_<test_type>`, e.g. `storage_unit`, `storage_single_thread`

But there is much work to do. I think there are other instances of this bug (maybe as many as 70).